### PR TITLE
Chitter Challenge

### PR DESCRIPTION
**Pull request by tcpickard94**

### CODE REVIEW

File: spec/web_helper.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  fill_in :password, with: password
+  click_button "Log in"
+end
+ 
+def create_peep
+  sign_up
+  fill_in :peep, with: "Hello, world!"
+  click_button 'Post'
+  click_button 'Log out'
+end
```
Comment (MattGough) [inline spec/web_helper.rb]: Could wrap methods in a module.

File: app/controllers/users.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    )
+    if @user.save 
+      session[:user_id] = @user.id
+      redirect '/home'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :'user/new'
+    end
+  end
+end
```
Comment (MattGough) [inline app/controllers/users.rb]: Could refactor to make controller methods shorter.

File: app/views/home.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                      <div class='reply'> 
+                              Replies: <br/>  
+                              <%= reply.body %>
+                              <%= reply.user.username %>
+                              <%= reply.formatted_time %>
+                      </div>
+                <% end %>
+        </div>
+<% end %>
+</main>
```
Comment (MattGough) [inline app/views/home.erb]: Could improve semantic HTML/CSS elements?

File: app/server.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  set :partial_template_engine, :erb
+ 
+  enable :partial_underscores 
+
+  helpers do
+    def current_user
+      @current_user = User.get(session[:user_id]) 
+    end
+  end
+end
```
Comment (MattGough) [inline app/server.rb]: Helper method could be extracted to helpers.rb file.

**Comment by mechacoach** [comment]
You had your chitter-challenge reviewed by **Matt Gough**

### The good points are:

* Correct installation instructions in README
* All unnecessary components removed, Ruby style guide conventions followed, useflash.nowandflash.nextcorrectly: flash.now and flash.next used correctly
* Rakefile has appropriate tasks
* Gemfile uses test groups
* Database Cleaner is set up correctly
* DataMapper set up in separate configuration file, ensurespechelper.rbisconfiguredcorrectly: spec_helper.rb pulls in entire app through single point, Only test helpers (not view helpers) are pulled in to spec_helper.rb
* Feature tests organized correctly
* Feature and unit tests in appropriate locations
* Spec 'Helpers' used appropriately
* Used modular style
* Routes split over separate controller files
* Forms use correct markup
* Correct use of Semantic HTML elements
* Partials used appropriately

### You should consider the following improvements:

* Ensure spec_helper.rb is Configured Correctly
* Ensure Asset Routes are Set Correctly
* Use flash.now and flash.next Correctly
* Refactor Long Controller Methods
* Use View Helpers

### Additional comments:
Really great, some nice styling and extra reply functionality!

see https://github.com/makersacademy/chitter-challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**